es6-persistence/src/main/java/com/netflix/conductor/es6/config/ElasticSearchV6Configuration.java

```diff
@@ -122,15 +125,29 @@ public RestClientBuilder restClientBuilder(ElasticSearchProperties properties) {
     public IndexDAO es6IndexRestDAO(
             RestClientBuilder restClientBuilder,
             ElasticSearchProperties properties,
+            @Qualifier("es6RetryTemplate") RetryTemplate retryTemplate,
             ObjectMapper objectMapper) {
-        return new ElasticSearchRestDAOV6(restClientBuilder, properties, objectMapper);
+        return new ElasticSearchRestDAOV6(
+                restClientBuilder, retryTemplate, properties, objectMapper);
     }
 
     @Bean
     @Conditional(IsTcpProtocol.class)
     public IndexDAO es6IndexDAO(
-            Client client, ElasticSearchProperties properties, ObjectMapper objectMapper) {
-        return new ElasticSearchDAOV6(client, properties, objectMapper);
+            Client client,
+            @Qualifier("es6RetryTemplate") RetryTemplate retryTemplate,
+            ElasticSearchProperties properties,
+            ObjectMapper objectMapper) {
+        return new ElasticSearchDAOV6(client, retryTemplate, properties, objectMapper);
+    }
+
+    @Bean(name = "es6RetryTemplate")
```

Comment:
This can also be replaced with `@Retrayable(backoff=@Backoff(delay=1000))` if prefer

Reply:
change method name to `es6RetryTemplate()`?

Reply:
The method to be retried is a private method and ideally should not be exposed. The annotation only works on public methods, hence I chose to use RetryTemplate. Open to change if you feel otherwise